tests/test_all_rules.py

```diff
@@ -696,3 +696,29 @@ def test_rule_backports_for_restricted_fields(self):
             err_msg = 'The following rules have min_stack_versions lower than allowed for restricted fields:\n'
             err_msg += invalid_str
             self.fail(err_msg)
+
+
+class TestBuildTimeFields(BaseRuleTest):
+    """Test validity of build-time fields."""
+
+    def test_build_fields_min_stack(self):
+        """Test that newly introduced build-time fields for a min_stack for applicable rules."""
+        invalids = []
+
+        for rule in self.production_rules:
+            min_stack = rule.contents.metadata.min_stack_version
```

Comment:
Will we never use soft-forking then, only hard-forking from the metadata field??

Reply:
based on our last discussions, I don't think it is supportable (or advisable)